python/lsst/daf/butler/datastores/fileLikeDatastore.py

```diff
@@ -0,0 +1,565 @@
+
+    def getStoredFileInfo(self, ref):
+        """Retrieve information associated with file stored in this
+        `Datastore`.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            The Dataset that is to be queried.
+
+        Returns
+        -------
+        info : `StoredFileInfo`
+            Stored information about this file and its formatter.
+
+        Raises
+        ------
+        KeyError
+            Dataset with that id can not be found.
+        """
+        record = self.records.get(ref.id, None)
+        if record is None:
+            raise KeyError("Unable to retrieve formatter associated with Dataset {}".format(ref.id))
+        # Convert name of StorageClass to instance
+        storageClass = self.storageClassFactory.getStorageClass(record.storage_class)
+        return StoredFileInfo(record.formatter, record.path, storageClass,
+                              checksum=record.checksum, size=record.file_size)
+
+    def _get_dataset_location_info(self, ref):
+        """Find the `Location` of the requested dataset in the
+        `Datastore` and the associated stored file information.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            Reference to the required `Dataset`.
+
+        Returns
+        -------
+        location : `Location`
+            Location of the dataset within the datastore.
+            Returns `None` if the dataset can not be located.
+        info : `StoredFileInfo`
+            Stored information about this file and its formatter.
+        """
+        # Get the file information (this will fail if no file)
+        try:
+            storedFileInfo = self.getStoredFileInfo(ref)
+        except KeyError:
+            return None, None
+
+        # Use the path to determine the location
+        location = self.locationFactory.fromPath(storedFileInfo.path)
+
+        return location, storedFileInfo
+
+    def _prepare_for_get(self, ref, parameters=None):
+        """Check parameters for ``get`` and obtain formatter and
+        location.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            Reference to the required Dataset.
+        parameters : `dict`
+            `StorageClass`-specific parameters that specify, for example,
+            a slice of the Dataset to be loaded.
+
+        Returns
+        -------
+        location : `Location`
+            The location from which to read the dataset.
+        formatter : `Formatter`
+            The `Formatter` to use to deserialize the dataset.
+        info : `StoredFileInfo`
+            Stored information about this file and its formatter.
+        assemblerParams : `dict`
+            Parameters to use for post-processing the retrieved
+            dataset.
+        component : `str`
+            The component to be retrieved (can be `None`).
+        readStorageClass : `StorageClass`
+            The `StorageClass` of the dataset being read.
```

Comment:
This is a lot of items to return in a tuple. A `dict`, a `SimpleNamespace`, or a new `dataclass` instance might be better, at least if this is going to be called in more than a couple of places.

Reply:
I was torn. They grew organically during the refactor and of course it's a bit of a red flag when you have to split the line that is calling this method before the equals sign because there are so many variables coming back. I was shying away from a dataclass or some struct thing because it seemed like a lot of overhead for what is an internal method, but I'll take another look at it.